src/lndk_offers.rs

```diff
@@ -508,8 +557,28 @@ impl InvoicePayer for Client {
             blinded_hops.push(new_hop);
         }
 
+        let introduction_node = match path.introduction_node {
+            IntroductionNode::NodeId(pubkey) => pubkey,
+            IntroductionNode::DirectedShortChannelId(direction, scid) => {
+                let get_info_request = ChanInfoRequest {
```

Comment:
Since we use the same or similar code up above for grabbing the node from the scid, lets split this off into a function?